### DISCUSSION TITLE
Publishing Diagrams Using Java

### DISCUSSION THREAD

**Discussion by VarmaCTS** [structurizr/java#279] [Q&A]
We are trying to use published Images from front End apart from Embed functionality. For this we need to publish Images manually in workspace UI. Can we do this publishing Images from Java, So that we can publish the images dynamically.
Any kind of help would be appreciated!
Thanks in advance.

**Comment by simonbrowndotje** [comment]
Diagrams are rendered client-side (i.e. in the web browser), so you can use headless Chrome and a Puppeteer script to automate that process; see https://github.com/structurizr/puppeteer for details.
